client/src/interface/AuthApiData.ts

```diff
@@ -1,9 +1,11 @@
+import { AnyARecord } from 'dns';
 import { User } from './User';
 
 export interface AuthApiDataSuccess {
   message: string;
   user: User;
   token: string;
+  profile: any;
```

Comment:
Maybe you can use the Profile interface here? Any profile fields which are not returned when a user is first created can be marked as optional.

Eg: 
`location?: string`